### PR TITLE
Avoid trying to retrieve the default ds if a named one should be used

### PR DESCRIPTION
**Pull request by jmartisk**
Fixes https://github.com/quarkiverse/quarkus-langchain4j/issues/667

### CODE REVIEW

File: embedding-stores/pgvector/runtime/src/main/java/io/quarkiverse/langchain4j/pgvector/runtime/PgVectorEmbeddingStoreRecorder.java [hunk 1]
```diff
@@ -19,10 +18,13 @@ public class PgVectorEmbeddingStoreRecorder {
     public Function<SyntheticCreationalContext<PgVectorEmbeddingStore>, PgVectorEmbeddingStore> embeddingStoreFunction(
             PgVectorEmbeddingStoreConfig config, String datasourceName) {
         return context -> {
-            AgroalDataSource dataSource = Optional.ofNullable(datasourceName)
-                    .map(DataSourceLiteral::new)
-                    .map(dl -> context.getInjectedReference(AgroalDataSource.class, dl))
-                    .orElse(context.getInjectedReference(AgroalDataSource.class, new Default.Literal()));
```
Comment (jmartisk) [inline embedding-stores/pgvector/runtime/src/main/java/io/quarkiverse/langchain4j/pgvector/runtime/PgVectorEmbeddingStoreRecorder.java:25]: This was a sneaky one... The problem here was that `context.getInjectedReference(AgroalDataSource.class, new Default.Literal())` is not wrapped a lambda expression, therefore it gets evaluated always...

Comment (geoand) [reply embedding-stores/pgvector/runtime/src/main/java/io/quarkiverse/langchain4j/pgvector/runtime/PgVectorEmbeddingStoreRecorder.java:25]: Ouch...

Comment (geoand) [reply embedding-stores/pgvector/runtime/src/main/java/io/quarkiverse/langchain4j/pgvector/runtime/PgVectorEmbeddingStoreRecorder.java:25]: One more reason to hate lambdas  @edeandrea

**Review by geoand — APPROVED** [review]
Thanks!

**Comment by geoand** [comment]
Any chance we can get a test?

**Comment by jmartisk** [comment]
Hmm, the tests in pgvector module are using `@QuarkusTest`. We will probably need to rewrite them to use `quarkus-junit5-internal` because they can't be mixed together.

**Comment by jmartisk** [comment]
I'll try that

**Comment by geoand** [comment]
🙏

File: embedding-stores/pgvector/runtime/src/main/java/io/quarkiverse/langchain4j/pgvector/runtime/PgVectorEmbeddingStoreRecorder.java (hunk 1 above)
Comment (edeandrea) [reply embedding-stores/pgvector/runtime/src/main/java/io/quarkiverse/langchain4j/pgvector/runtime/PgVectorEmbeddingStoreRecorder.java:25]: Thanks for the zing @geoand :) I'll still keep my stance that, generally speaking, functional code is still easier to read and figure out what's going on than imperative code. 

That being said, in this particular case, one has to understand the subtle differences between `Optional.orElse` and `Optional.orElseGet`.

**Comment by geoand** [comment]
Indeed they don't AFAIR

**Comment by jmartisk** [comment]
We can work around it by temporarily extending `EmbeddingStoreIT` instead of `EmbeddingStoreWithFilteringIT`, then we won't have any parametrized tests, but we would lose test coverage of metadata filtering...

**Comment by geoand** [comment]
Up to you

**Review by geoand — APPROVED** [review]

**Pull request merged**